src/transformers/models/swin/modeling_swin.py

```diff
@@ -0,0 +1,915 @@
+            self.shift_size = 0
+            self.window_size = min(self.input_resolution)
+
+        self.layernorm_before = nn.LayerNorm(dim, eps=config.layer_norm_eps)
+        self.attention = SwinAttention(config, dim, num_heads)
+        self.drop_path = DropPath(config.drop_path_rate) if config.drop_path_rate > 0. else nn.Identity() 
+        self.layernorm_after = nn.LayerNorm(dim, eps=config.layer_norm_eps)  
+        self.intermediate = SwinIntermediate(config, dim)
+        self.output = SwinOutput(config, dim)
+
+        if self.shift_size > 0:
+            # calculate attention mask for SW-MSA
+            height, width = self.input_resolution
+            img_mask = torch.zeros((1, height, width, 1))
+            h_slices = (slice(0, -self.window_size),
+                        slice(-self.window_size, -self.shift_size),
+                        slice(-self.shift_size, None))
+            w_slices = (slice(0, -self.window_size),
+                        slice(-self.window_size, -self.shift_size),
+                        slice(-self.shift_size, None))
+            cnt = 0
+            for h in h_slices:
+                for w in w_slices:
+                    img_mask[:, h, w, :] = cnt
+                    cnt += 1
+
+            mask_windows = window_partition(img_mask, self.window_size) 
+            mask_windows = mask_windows.view(-1, self.window_size * self.window_size)
+            attn_mask = mask_windows.unsqueeze(1) - mask_windows.unsqueeze(2)
+            attn_mask = attn_mask.masked_fill(attn_mask != 0, float(-100.0)).masked_fill(attn_mask == 0, float(0.0))
+        else:
+            attn_mask = None
+
+        self.attn_mask = attn_mask    
+
+    def forward(
+        self,
+        hidden_states,
+        head_mask=None,
+        output_attentions=False,
+    ):
+        height, width = self.input_resolution
+        batch_size, dim, channels = hidden_states.size()
+        shortcut = hidden_states
+
+        hidden_states = self.layernorm_before(hidden_states)
+        hidden_states = hidden_states.view(batch_size, height, width, channels)
+
+        # cyclic shift
+        if self.shift_size > 0:
+            shifted_hidden_states = torch.roll(hidden_states, shifts=(-self.shift_size, -self.shift_size), dims=(1, 2))
+        else:
+            shifted_hidden_states = hidden_states
+
+        # partition windows
+        hidden_states_windows = window_partition(shifted_hidden_states, self.window_size)  
+        hidden_states_windows = hidden_states_windows.view(-1, self.window_size * self.window_size, channels)  
+
+        self_attention_outputs = self.attention(
+            hidden_states_windows,
+            self.attn_mask,
+            head_mask,
+            output_attentions=output_attentions,
+        )
+
+        attention_output = self_attention_outputs[0]
+
+        # if decoder, the last output is tuple of self-attn cache
```

Comment:
```suggestion
```
This comment can be deleted, as the model cannot be used as an autoregressive decoder.